server/api/v1/environment_service.go

```diff
@@ -145,6 +167,18 @@ func (s *EnvironmentService) DeleteEnvironment(ctx context.Context, request *v1p
 		return nil, status.Errorf(codes.InvalidArgument, "environment %q has been deleted", environmentID)
 	}
 
+	// All instances in the environment must be deleted.
+	instances, err := s.store.ListInstancesV2(ctx, &store.FindInstanceMessage{
```

Comment:
We can implement the CountInstance func here?

Reply:
Good suggestion, done.